### PR TITLE
chore: bzlmod support

### PR DESCRIPTION
**Pull request by alexeagle**

### CODE REVIEW

**Review by adzenith — APPROVED** [review]

**Comment by alexeagle** [comment]
Sorry, just a draft to get a CI status - needs a bunch more work...

**Comment by adzenith** [comment]
I'm also happy to push to this PR to get it green if you like. I have some bzlmod-ifications for this that I've been meaning to bundle up.

**Comment by alexeagle** [comment]
Thanks, I had it partly done last night though - I think it's ready.

File: .github/workflows/continuous-integration.yml [hunk 1]
```diff
@@ -9,11 +9,31 @@ on:
...
     runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        bzlmodEnabled: [true, false]
+        folder: [".", "examples"]
+        exclude:
+          # No point keeping the root workspace compatible with --noenable_bzlmod
+          - folder: "."
+            bzlmodEnabled: false
```
Comment (adzenith) [inline .github/workflows/continuous-integration.yml:20]: did you want this to be `true`?

Comment (alexeagle) [reply .github/workflows/continuous-integration.yml:20]: this is `false` to exclude the `folder: ., bzlmodEnabled: false` entry from the matrix. It's a double negative :)

File: third_party/BUILD [hunk 2]
```diff
@@ -1,4 +1,4 @@
-load("@mypy_integration_pip_deps//:requirements.bzl", "requirement")
+load("@pypi//:requirements.bzl", "requirement")
```
Comment (f0rmiga) [inline third_party/BUILD:1]: I think changing the name of the pip deps repository to `pypi` will break other repositories that set the same for their internal deps. Could you undo this, please?

Comment (alexeagle) [reply third_party/BUILD]: yeah, I agree

**Pull request merged**